app/models/namespace.rb

```diff
@@ -177,4 +181,21 @@ def create_delete_activities!(actor)
       }
     )
   end
+
+  def self.create_from_name!(name)
```

Comment:
If the namespace didn't persist it should return nil. Note that callers from `self.get_from_repository_name` already expect this kind of behavior.

Reply:
Moreover, wouldn't it be easier to simply use the create service as we do on the API?

Reply:
> If the namespace didn't persist it should return nil. Note that callers from `self.get_from_repository_name` already expect this kind of behavior.

It's already returning, no? I've changed it to `create` instead of `create!` if that was your concern.

> Moreover, wouldn't it be easier to simply use the create service as we do on the API?

We can reuse that instead of creating the activity manually.

Reply:
The thing is that with `create`, you got a namespace, even if not persisted. Therefore, this method would never return nil, but just a flawed `namespace` object. That was my issue.

Now, if you use `create!`, it will actually raise an exception if something wrong happens. This is not handled neither here nor in `get_from_repository_name`. Since in the `get_from_repository_name` method the whole logic already considers nil namespaces, I think we should stick to that. Thus, I would keep using `create` here. If it didn't persist, then I'd return nil (and possibly log a debug message). Something like:

```ruby
namespace = Namespace.create!(
    name:       name,
    registry:   Registry.get,
    visibility: Namespace.visibilities[:visibility_private],
    team:       Team.global
)
return unless namespace.persisted?

# create activity and return
```

Also note that the create service also follows this logic :wink: